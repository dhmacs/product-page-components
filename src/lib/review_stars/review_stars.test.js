import React from 'react';
import ReviewStars from './review_stars';
import renderer from 'react-test-renderer';

Math.random = jest.fn(() => 1);

it('renders correctly', () => {
   const tree = renderer
      .create(<ReviewStars numReviews={30} average={3.2} starSize={18} />)
      .toJSON();
   expect(tree).toMatchSnapshot();
});
