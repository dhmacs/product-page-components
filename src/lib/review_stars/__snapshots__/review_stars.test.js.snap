// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="sc-bdVaJa kWUKBK"
  href="#productReviews"
>
  <div
    className="star-ratings"
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "position": "relative",
      }
    }
    title="3.2 Stars"
  >
    <svg
      className="star-grad"
      style={
        Object {
          "height": "0",
          "position": "absolute",
          "visibility": "hidden",
          "width": "0",
          "zIndex": "0",
        }
      }
    >
      <defs>
        <linearGradient
          id="starGrad000000000000000"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        >
          <stop
            className="stop-color-first"
            offset="0%"
            style={
              Object {
                "stopColor": "#0071CE",
                "stopOpacity": "1",
              }
            }
          />
          <stop
            className="stop-color-first"
            offset="20%"
            style={
              Object {
                "stopColor": "#0071CE",
                "stopOpacity": "1",
              }
            }
          />
          <stop
            className="stop-color-final"
            offset="20%"
            style={
              Object {
                "stopColor": "#E1E1E2",
                "stopOpacity": "1",
              }
            }
          />
          <stop
            className="stop-color-final"
            offset="100%"
            style={
              Object {
                "stopColor": "#E1E1E2",
                "stopOpacity": "1",
              }
            }
          />
        </linearGradient>
      </defs>
    </svg>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": undefined,
          "paddingRight": "0px",
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg"
        style={
          Object {
            "height": "18px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "18px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "#0071CE",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg"
        style={
          Object {
            "height": "18px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "18px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "#0071CE",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg"
        style={
          Object {
            "height": "18px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "18px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "#0071CE",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg multi-widget-selected"
        style={
          Object {
            "height": "18px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "18px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "url('#starGrad000000000000000')",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": "0px",
          "paddingRight": undefined,
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg"
        style={
          Object {
            "height": "18px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "18px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "#E1E1E2",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
  </div>
  <h4
    className="sc-bwzfXH jGfGgN"
  >
    30
     
    reviews
  </h4>
</a>
`;
